File: fis-nutricion-infantil/backend/src/main/java/backend/service/RecetaService.java
package backend.service;

import backend.model.Receta;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

@Service
public class RecetaService {
    private final List<Receta> recetas = new ArrayList<>();

    public Receta guardar(Receta receta) {
        recetas.add(receta);
        return receta;
    }

    public List<Receta> listar() {
        return recetas;
    }

    public List<Receta> consultarRecetarioPersonalizado() {
        return recetas.stream()
                .filter(Receta::validarRestricciones)
                .toList();
    }

    public List<Receta> recomendarPorAporteCalorico(double caloriasMaximas) {
        return recetas.stream()
                .filter(receta -> receta.obtenerAporteCalorico() <= caloriasMaximas)
                .toList();
    }

    public Optional<Receta> buscarPorIndice(int indice) {
        if (indice < 0 || indice >= recetas.size()) {
            return Optional.empty();
        }
        return Optional.of(recetas.get(indice));
    }
}

File: fis-nutricion-infantil/frontend/src/services/recetaService.js
import apiClient from './apiClient'

export const recetaService = {
  listar: () =>
    apiClient.get('/recetas').then((response) => response.data),

  guardar: (data) =>
    apiClient.post('/recetas', data).then((response) => response.data),

  personalizadas: () =>
    apiClient.get('/recetas/personalizadas').then((response) => response.data),

  recomendadasPorCalorias: (caloriasMaximas) =>
    apiClient
      .get('/recetas/recomendadas', { params: { caloriasMaximas } })
      .then((response) => response.data),

  buscarPorIndice: (indice) =>
    apiClient.get(`/recetas/${indice}`).then((response) => response.data),
}